pandas/core/groupby.py

```diff
@@ -2586,10 +2586,21 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     """
     group_axis = obj._get_axis(axis)
 
-    # validate that the passed level is compatible with the passed
+    # validate that the passed single level is compatible with the passed
     # axis of the object
     if level is not None:
-        if not isinstance(group_axis, MultiIndex):
+        # TODO: These two conditions are almost same.
```

Comment:
ok for now. can you come back in a future PR and see what we can do with all the conditions in this section. getting pretty unweildy (and document as much as possible).

Reply:
@jreback Okay. I'll do that. These are too complicated to do refactoring in this PR, I think.